src/inmanta/agent/handler.py

```diff
@@ -179,6 +179,30 @@ def __init__(
         else:
             self.logger = logger
 
+        self._facts: List[Dict[str, Any]] = []
+
+    def set_fact(self, fact_id: str, value: str, metadata: dict = {}) -> None:
+        """
+            Send a fact to the Inmanta server.
+
+            :param fact_id: The name of the fact.
+            :param value: The actual value of the fact.
+            :param metadata: Metadata about the fact.
+        """
+        resource_id = self._resource.id.resource_str()
+        fact = {
+            "id": fact_id,
+            "source": "agent",
```

Comment:
There on line 1013 of inmanta/data/__init__.py an unused list of all types of sources 

```
SOURCE = ("fact", "plugin", "user", "report")
```
Perhaps we have to update this. 

I don't know what the effect will be of setting the source to agent instead of fact.
But as far as I can see, this is going to be OK.

Reply:
This is used in the dashboard and I think also in std::getfact, so watch out with given facts a different source.